product_brand_stock_account/models/stock_valuation_layer.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Francesco Apruzzese <cescoap@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockValuationLayer(models.Model):
+
+    _inherit = "stock.valuation.layer"
+
+    product_brand_id = fields.Many2one(
+        comodel_name="product.brand",
+        related="product_id.product_brand_id",
+        string="Brand",
+        store=True,
```

Comment:
we need a pre_init_hook to initialize this column

Reply:
I have performed several tests. When installing the module, if the products already have a brand associated with it, this column is populated when it is created during installation. So the script is not needed.